Include/internal/pycore_interp.h

```diff
@@ -168,8 +171,26 @@ struct _is {
 
     struct ast_state ast;
     struct type_cache type_cache;
+
+    // This must be last.
+    struct {
+        int initialized;
+        PyThreadState tstate;
+        // XXX Pre-allocate as many objects as possible here.
```

Comment:
This may not be true. If we go for immortal objects, then will be no need for per-interpreter pre-allocated objects.
So either remove or clarify this comment.

Reply:
This comment refers to all the object pointers on `PyInterpreterState` (e.g. "modules", "slice_cache").  They will remain per-interpreter.  I'll clarify the comment.

Also, regardless of if they are immortal or not, those objects still need to be allocated.  So we can pre-allocate them to avoid allocation at runtime and to simplify the init code.